File: dot-net-projects/vbasp-project-log/VBAspProjectLog/Controllers/LoginController.vb
Imports System.Web.Mvc
Imports System.Web.Mvc.HttpPostAttribute

Public Class LoginController
    Inherits System.Web.Mvc.Controller

    Private db As New ProjectLogContext()

    '
    ' GET: /Login
    Function Index() As ActionResult
        'This is the main login page where users go to enter credentials.
        Return View()
    End Function

    '
    ' POST: /Login
    <HttpPost()> _
    Function Index(ByVal action As String) As ActionResult

        If action = "logout" Then
            Session("UserIsAuthorized") = False
            Session.Remove("Username")
        End If

        Return View()
    End Function

    <HttpPost()> _
    Function Login(ByVal username As String, ByVal password As String) As ActionResult

        'Authenticate user via database
        Dim qUser As User = (From u In db.Users
                           Where u.UserName = username
                           Select u).FirstOrDefault()

        Dim authenticated As Boolean = False

        If (Not qUser Is Nothing) Then
            If password = qUser.Password Then
                Session.Add("UserIsAuthorized", True)
                Session.Add("Username", username)
                authenticated = True
            End If
        End If

        If authenticated Then Return RedirectToAction("Index", "Home")

        TempData("Message") = "The username and/or password were not recognised."

        Return RedirectToAction("Index")

    End Function

End Class

File: dot-net-projects/csasp-lms/UniversityLMS/Controllers/LoginController.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Data;
using System.Diagnostics;
using System.Web.Mvc;
using UniversityLMS.DAL;

public class LoginController : System.Web.Mvc.Controller
{
    private LMSContext db = new LMSContext();
    //
    // GET: /Login
    public ActionResult Index()
    {
        //This is the main login page where users go to enter credentials.
        return View();
    }

    //
    // POST: /Login
    [HttpPost()]
    public ActionResult Index(string action)
    {
        if (action == "logout")
        {
            Session["UserIsAuthorized"] = false;
            Session.Remove("Username");
        }

        return View();
    }

    [HttpPost()]
    public ActionResult Login(string username, string password)
	{

		//Authenticate user via database
		User qUser = (from u in db.Users where u.UserName == username select u).FirstOrDefault();

		bool authenticated = false;

		if (((qUser != null))) {
			if (password == qUser.Password) {
				Session.Add("UserIsAuthorized", true);
				Session.Add("Username", username);
				authenticated = true;
			}
		}

		if (authenticated)
			return RedirectToAction("Index", "Home");

		TempData["Message"] = "The username and/or password were not recognised.";
        
		return RedirectToAction("Index");
        
	}

}
